mautic/mautic
# Fix for php-cs-fixer pre-commit issue when running Windows

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | 
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | #3155 and #3145 
| BC breaks? | 
| Deprecations? | 

#### Description:
This PR addresses issues #3145 and #3155 and ensures windows developers comply with the same coding standards as nix developers when committing using git.

This fix also helps to improve the first impression of Mautic in 2 situations.
1. Windows Developers installing Mautic for the first time
2. Running `composer install / update` using the Release zip. (folders missing)

In both these situations, at the end of the process there is an error because the current line only works in a nix environment!
```
cp ./build/hooks/pre-commit ./.git/hooks/pre-commit
```

By simply changing 2 lines of the pre-commit script, for windows, and using php to complete the final step we can:

1. run on all platforms
2. determine what OS is running
3. test if everything is where it's supposed to be
4. copy the appropriate file. Ensures coding standards on commits
5. eliminate unnecessary confusion and potential support issues. (no more Error)

**Steps to test this PR:**

Install Mautic using git into a test folder (`git clone https://github.com/mautic/mautic.git`)
Apply this PR
run composer install
There should be no error (if running Windows)

The following should now work on all platforms...

Change `offline.php` and add the following.
```
$aTest = array();
```
Add your changes and commit. `git add .` and `git commit -m "Test pre-commit"` you should see the output from php-cs-fixer. e.g.
```
C:\Test\mautic>git commit -v --branch --status -m "Testing Composer Mods with git Commit triggers"
php-cs-fixer pre commit hook start
Loaded config from "C:/Test/mautic/.php_cs".
F
Legend: ?-unknown, I-invalid file syntax, file ignored, .-no changes, F-fixed, E-error
   1) C:\Test\mautic\offline.php (short_array_syntax, unalign_equals, align_equals, braces)
Fixed all files in 2.314 seconds, 6.000 MB memory used
php-cs-fixer pre commit hook finish
[composer-test e8b5f81] Testing Composer Mods with git Commit triggers
 1 file changed, 2 insertions(+), 3 deletions(-)
```

Inspection should show the inserted line now has `$aTest = [];`
 
**Steps to reproduce the bug:**
See #3145 and #3155
